stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -583,42 +597,117 @@ function complete_any_methods(ex_org::Expr, callee_module::Module, context_modul
     return out
 end
 
+
+function detect_invalid_kwarg!(kwargs_ex, n, kwargs_flag, allow_splat)
+    if n isa Symbol
+        push!(kwargs_ex, n)
+        return kwargs_flag
+    end
+    allow_splat && isexpr(n, :...) && return kwargs_flag
+    return 2 # The kwarg is invalid
+end
+
 function complete_methods_args(funargs::Vector{Any}, ex_org::Expr, context_module::Module, default_any::Bool, allow_broadcasting::Bool)
     args_ex = Any[]
-    kwargs_ex = false
+    kwargs_ex = Symbol[]
+    kwargs_flag = 0
+    # kwargs_flag is:
+    # * 0 if there is no semicolon and no invalid kwarg
+    # * 1 if there is a semicolon and no invalid kwarg
+    # * 2 if there are two semicolons or more, or if some kwarg is invalid, which
+    #        means that it is not of the form "bar=foo", "bar" or "bar..."
     if allow_broadcasting && ex_org.head === :. && ex_org.args[2] isa Expr
         # handle broadcasting, but only handle number of arguments instead of
         # argument types
-        for _ in (ex_org.args[2]::Expr).args
-            push!(args_ex, Any)
+        for ex in (ex_org.args[2]::Expr).args
+            if isexpr(ex, :parameters)
+                kwargs_flag = ifelse(kwargs_flag == 0, 1, 2) # there should be at most one :parameters
+                for x in ex.args
+                    n = isexpr(x, :kw) ? x.args[1] : x
+                    if !(n isa Symbol) || !isexpr(x, :...)
+                        kwargs_flag = 2
+                    end
```

Comment:
I ended up refactoring `complete_methods_args` as well since the kwarg logic was entirely shared between the two branches (with and without broadcasting)